samples/examples/FSharpExamples/Matrix.fs

```diff
@@ -28,38 +28,53 @@ open Prajna.Api.FSharp
 
 open Prajna.Examples.Common
 
-type MatrixSample() =
+module Matrix =
 
     let (.*) (xs: float[]) (ys: float[]) = 
-        // simulate doing some more work, 
-        // e.g.: run the remaining layers of a neural net
-        Threading.Thread.Sleep(3) 
         Array.fold2 (fun sum x y -> sum + x * y) 0.0 xs ys 
 
     let printTiming (msg: string) (x: Lazy<'b>) : 'b =
         let sw = Stopwatch.StartNew()
+        printf "%s: " msg
         let ret = x.Value
         sw.Stop()
-        printfn "%s: %A" msg sw.Elapsed
+        printfn "%A" sw.Elapsed
         ret
 
-    let run (cluster: Cluster) =
+    let random = new Random(0)
+
+    let randomVector n =  Array.init n (fun _ -> random.NextDouble())
+    
+    let randomMatrix m n =  Array.init m (fun _ -> randomVector n)
+
+    /// Matrix-Vector multiplication for illustrational purposes.
+    /// The overhead of distributing makes it faster to run this locally.
+    /// See Matrix-Matrix for a case where the computation is actually faster distributed.
+    let matrixVector (cluster: Cluster) =
 
         let numRows = 200
         let vectorSize = 100
 
-        let r = new Random(0)
-        let randomMatrix =  Array.init numRows (fun _ -> Array.init vectorSize (fun _ -> r.NextDouble()))
-        let vector = Array.init vectorSize (fun _ -> r.NextDouble())
+        let matrix =  randomMatrix numRows vectorSize
+        let vector = randomVector vectorSize
 
         let singleCoreResult = 
             printTiming "single core" <| lazy 
-                randomMatrix |> Array.map (fun row -> row .* vector)
-        
+                matrix |> Array.map (fun row -> row .* vector)
+
+        let numPartitions = Environment.ProcessorCount
```

Comment:
The sample is meant to run locally. Ideally, it would be Environment.ProcessorCount / cluster.NumNodes, but this keeps things simple without hurting speed too much. I should add a comment that on an actual cluster, we would do something different.

From: Weirong Zhu [mailto:notifications@github.com]
Sent: Friday, October 2, 2015 5:44 PM
To: MSRCCS/Prajna <Prajna@noreply.github.com>
Cc: Bruno Bozza <brunosb@microsoft.com>
Subject: Re: [Prajna] Adding Matrix-Matrix sample, updating Matrix-Vector (#68)


In samples/examples/FSharpExamples/Matrix.fs<https://na01.safelinks.protection.outlook.com/?url=https%3a%2f%2fgithub.com%2fMSRCCS%2fPrajna%2fpull%2f68%23discussion_r41081287&data=01%7c01%7cbrunosb%40microsoft.com%7cba0f64e6401c491a68f408d2cb8bca39%7c72f988bf86f141af91ab2d7cd011db47%7c1&sdata=bDXVaWRNPb%2fsyEp70bxSeinUHPkF3XCNNSHGiznjqUc%3d>:

>

>          let singleCoreResult =

>              printTiming "single core" <| lazy

> -                randomMatrix |> Array.map (fun row -> row .* vector)

> -

> +                matrix |> Array.map (fun row -> row .* vector)

> +

> +        let numPartitions = Environment.ProcessorCount

Why use local machine's processor count to determine the number of partitions for each remote node?

—
Reply to this email directly or view it on GitHub<https://na01.safelinks.protection.outlook.com/?url=https%3a%2f%2fgithub.com%2fMSRCCS%2fPrajna%2fpull%2f68%2ffiles%23r41081287&data=01%7c01%7cbrunosb%40microsoft.com%7cba0f64e6401c491a68f408d2cb8bca39%7c72f988bf86f141af91ab2d7cd011db47%7c1&sdata=vqdwcfbnnQcTqu4%2f2bYRKmdeXLRNjuK39y4yeXQB3ks%3d>.